File: Android/BarkCloud.Android/app/src/main/java/com/barkfluff/BarkCloud/ui/main/MainDestination.kt
package com.barkfluff.BarkCloud.ui.main

import androidx.annotation.StringRes
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Folder
import androidx.compose.material.icons.filled.PhotoAlbum
import androidx.compose.material.icons.filled.PhotoLibrary
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.outlined.Delete
import androidx.compose.material.icons.outlined.Folder
import androidx.compose.material.icons.outlined.PhotoAlbum
import androidx.compose.material.icons.outlined.PhotoLibrary
import androidx.compose.material.icons.outlined.Settings
import androidx.compose.ui.graphics.vector.ImageVector
import com.barkfluff.BarkCloud.R

/**
 * Пять вкладок нижней навигации — точь-в-точь как в iOS-клиенте.
 * Порядок = порядок в таб-баре. По умолчанию открывается «Альбомы».
 * [route] — это маршрут вложенного навигационного графа вкладки.
 */
enum class MainDestination(
    val route: String,
    @StringRes val labelRes: Int,
    val iconOutlined: ImageVector,
    val iconFilled: ImageVector,
) {
    Gallery(
        route = "gallery",
        labelRes = R.string.tab_gallery,
        iconOutlined = Icons.Outlined.PhotoLibrary,
        iconFilled = Icons.Filled.PhotoLibrary,
    ),
    Files(
        route = "files",
        labelRes = R.string.tab_files,
        iconOutlined = Icons.Outlined.Folder,
        iconFilled = Icons.Filled.Folder,
    ),
    Albums(
        route = "albums",
        labelRes = R.string.tab_albums,
        iconOutlined = Icons.Outlined.PhotoAlbum,
        iconFilled = Icons.Filled.PhotoAlbum,
    ),
    Trash(
        route = "trash",
        labelRes = R.string.tab_trash,
        iconOutlined = Icons.Outlined.Delete,
        iconFilled = Icons.Filled.Delete,
    ),
    Settings(
        route = "settings",
        labelRes = R.string.tab_settings,
        iconOutlined = Icons.Outlined.Settings,
        iconFilled = Icons.Filled.Settings,
    );

    companion object {
        val Default = Albums
    }
}

File: Ios/BarkCloud/BarkCloud/Features/Main/MainDestination.swift
import Foundation
import SwiftUI

/// Пять вкладок нижней навигации. Порядок = порядок в таб-баре.
/// По умолчанию открывается «Альбомы».
enum MainDestination: Hashable, CaseIterable {
    case gallery   // локальная медиатека устройства (PhotoKit)
    case files     // файлы: устройство + облако + общие
    case albums    // облачные медиа: Фото / Видео / Альбомы
    case trash     // корзина облака
    case settings  // профиль и настройки

    static let `default`: MainDestination = .albums

    var labelKey: LocalizedStringResource {
        switch self {
        case .gallery: return "tab_gallery"
        case .files: return "tab_files"
        case .albums: return "tab_albums"
        case .trash: return "tab_trash"
        case .settings: return "tab_settings"
        }
    }

    var iconOutlined: String {
        switch self {
        case .gallery: return "photo.on.rectangle"
        case .files: return "folder"
        case .albums: return "rectangle.stack"
        case .trash: return "trash"
        case .settings: return "gearshape"
        }
    }

    var iconFilled: String {
        switch self {
        case .gallery: return "photo.on.rectangle.fill"
        case .files: return "folder.fill"
        case .albums: return "rectangle.stack.fill"
        case .trash: return "trash.fill"
        case .settings: return "gearshape.fill"
        }
    }
}
